### PR TITLE
WD-10180 - feature: add panel for creating roles

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add panel for creating roles.
- Update API spec to match the backend.
- Set auth headers.

## QA

- Set `VITE_DEMO_AUTH_TOKEN` to a base64 version of an administrator's username, see [the docs](https://github.com/canonical/identity-platform-admin-ui/wiki/OpenFGA-setup#api-authorization).
- Go to the roles page.
- Click "Create role"
- Fill in the name and submit.
- The form should close and the table should update with the new role.

## Details

- https://warthogs.atlassian.net/browse/WD-10180

## Screenshots

Screenshot 2024-04-09 at 4 03 16 PM

### CODE REVIEW

File: src/components/ToastCard/ToastCard.test.tsx
Comment (huwshimi) [inline src/components/ToastCard/ToastCard.test.tsx:1]: This component and tests etc. were copied over from Juju Dashboard.

**Review by vladimir-cucu — APPROVED** [review]
Looks good and works as expected locally! :rocket: If you think it might be worth it, maybe we could improve slightly the logic in `useEffect` in `ReBACAdmin.tsx` and remove some of the `expect(...).toBeInTheDocument()` in `test/utils.tsx`.

File: openapi.yaml [hunk 1]
```diff
@@ -594,16 +594,14 @@ paths:
         content:
           application/json:
             schema:
-              $ref: '#/components/schemas/Role'
+              $ref: '#/components/schemas/RoleObject'
       responses:
         '200':
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply openapi.yaml:599]: Thanks! Fixed.

**Comment by huwshimi** [comment]
> Looks good and works as expected locally! 🚀 If you think it might be worth it, maybe we could improve slightly the logic in `useEffect` in `ReBACAdmin.tsx` and remove some of the `expect(...).toBeInTheDocument()` in `test/utils.tsx`.

I've done these and added tests for the useEffect.

**Pull request merged**